controllers/topics/create.js

```diff
@@ -0,0 +1,42 @@
+const {Topics} = require('../../databases/models');
+
+module.exports = async (req, res) => {
+  try {
+    let {name} = req.body;
+    name = name.toLowerCase();
+    let topics = await Topics.findOne({
+      where: {
+        name,
+        deleted_at: null
+      }
+    });
+
+    if (topics) {
+      res.status(400).json({
+        code: 400,
+        status: 'failed',
+        message: 'Topic already exists'
+      });
+    }
+
+    await Topics.create({
+      name,
+      icon_path: '',
+      is_custom_topic: true,
+      created_at: new Date(),
+      categories: ''
+    });
+
+    return res.status(200).json({
+      success: true,
+      message: 'Success'
+    });
+  } catch (error) {
+    return res.json({
+      code: error.statusCode,
+      status: 'fail',
+      message: error.message,
+      data: 'null'
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling could be improved here. The `statusCode` property is not always present on all errors. It would be better to check if it exists before using it. If it doesn't exist, you can default to a generic server error status code like 500. Also, returning a string 'null' for data is not a good practice. It's better to return null without quotes or simply omit the data field if there's no data to return.

```diff
    return res.json({
-36:       code: error.statusCode,
+36:       code: error.statusCode || 500,
      status: 'fail',
      message: error.message,
-39:       data: 'null'
+39:       data: null
    });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->